src/main/java/org/zalando/nakadi/controller/SubscriptionController.java

```diff
@@ -115,12 +119,13 @@ public SubscriptionController(final SubscriptionDbRepository subscriptionReposit
         return status(HttpStatus.CREATED).body(subscription);
     }
 
-    private List<String> checkExistingEventTypes(final SubscriptionBase subscriptionBase)
+    private List<String> checkExistingEventTypes(final SubscriptionBase subscriptionBase, final Client client)
             throws InternalNakadiException {
         final List<String> noneExistingEventTypes = Lists.newArrayList();
         for (final String etName : subscriptionBase.getEventTypes()) {
             try {
-                eventTypeRepository.findByName(etName);
+                final EventType eventType = eventTypeRepository.findByName(etName);
+                client.checkScopes(eventType.getReadScopes());
```

Comment:
The name of the method is `checkExistingEventTypes` but there we are doing authorisation. I think it should go elsewhere. I know it's handy here because we just got the event type from the repository, but I would rather refactor it to have it in a more appropriate place.